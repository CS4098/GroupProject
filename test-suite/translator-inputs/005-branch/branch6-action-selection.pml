process test{
	branch br1 {
		action act1 {
		requires { a }
		provides { e }
		}
		selection sel1 {
	        action act2 {
		        requires { a }
		        provides { b }
	        }
	        action act3 {
		        requires { b }
		        provides { e }
	        }
	        action act4 {
		        requires { c }
		        provides { d }
	        }
    	}
	}
}
