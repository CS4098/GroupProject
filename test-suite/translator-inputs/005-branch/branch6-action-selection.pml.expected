bool a;
bool b;
bool c;
bool d;
bool e;

active proctype test()
{
	int br1 = _nr_pr;
	run act1();
	run sel1();
	_nr_pr == br1 ->
}

proctype act1()
{
	a ->
		e = true;
}

proctype sel1()
{
    if
    :: true ->
             a ->
                b = true;
    :: true ->
             b ->
                e = true;
    :: true ->
             c ->
                d = true;
    fi
}
